### PR TITLE
Facelift/title cherrypick

### PR DESCRIPTION
**Pull request by ninaandal**
### Description
Cherrypicked from Binoy's great work on the document title 🏅
Have added a maximum of 4 lines to the title height.

### What to review
The new title headers in the documents

### CODE REVIEW

**Review by binoy14 — APPROVED** [review]
Changes looks good to me!

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  $muted?: boolean
+  theme: Theme
+}
+export const Title = styled.h6<TitleProps>`
+  ${({$titleHeight, $muted, theme}) => {
+    const {lineHeight} = theme.sanity.fonts.heading.sizes[5]
+
+    const fontSizeSmall = theme.sanity.fonts.heading.sizes[3].fontSize
+    const fontSizeMedium = theme.sanity.fonts.heading.sizes[4].fontSize
+    const fontSizeLarge = theme.sanity.fonts.heading.sizes[5].fontSize
```
Comment (ninaandal) [inline packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts:18]: Feel free to change the sizes of these if we want them to be smaller/bigger!

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  position: relative;
+`
+
+export const Title = styled(Heading)`
+  ${({theme}) => {
+    const fontSizeSmall = theme.sanity.fonts.heading.sizes[3].fontSize
+    const fontSizeLarge = theme.sanity.fonts.heading.sizes[4].fontSize
+
+    return css`
+      overflow-wrap: break-word;
```
Comment (robinpyon) [inline packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts:14]: Changed from `word-break: break-word;` as break-word is [now deprecated](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#break-word)

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import styled, {css} from 'styled-components'
+
+export const Root = styled(Box)`
+  position: relative;
+`
+
+export const Title = styled(Heading)`
+  ${({theme}) => {
+    const fontSizeSmall = theme.sanity.fonts.heading.sizes[3].fontSize
+    const fontSizeLarge = theme.sanity.fonts.heading.sizes[4].fontSize
```
Comment (robinpyon) [inline packages/sanity/src/desk/panes/document/documentPanel/documentViews/styles.ts:11]: Changed from `theme.sanity.fonts.heading.sizes[5].fontSize`, just to make the jump between font sizes less severe

File: packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx [hunk 4]
```diff
@@ -152,6 +155,14 @@ export const FormView = forwardRef<HTMLDivElement, FormViewProps>(function FormV
     >
       <PresenceOverlay margins={margins}>
         <Box as="form" onSubmit={preventDefault} ref={setRef}>
+          {/* This is the document title */}
+          <Box marginBottom={5}>
+            <ElementQuery>
+              <Title forwardedAs="h2" muted={!title}>
```
Comment (robinpyon) [inline packages/sanity/src/desk/panes/document/documentPanel/documentViews/FormView.tsx:161]: Changed from `forwardedAs="h6"`, as this title is now much more prominent (but not marked as `h1` as there can be multiple instances of these at any one time)

**Review by robinpyon — APPROVED** [review]
Thank you both!

@ninaandal made a few minor changes, comments below

**Pull request merged**